### ISSUE TITLE
💥 Crash on  ` -[PNDNSURLSecureSession isTokenExist]`

### ISSUE THREAD

**Issue by msrutek-paylocity** [pendo-io/pendo-mobile-sdk#148]
## **Platform + Version**
iOS 17.5.1
iPhone 12

## **SDK Version**
3.2.2

## **Framework**
Native, SwiftUI

## **Describe the bug**
Crash of the app.

## **To Reproduce**
Yet unknown.

## **Expected behavior**
No crash.

## Stack Trace

```
Crashed: com.apple.root.user-initiated-qos

0  libobjc.A.dylib                0xb590 objc_autoreleaseReturnValue + 76
1  Pendo                          0x13a7c -[PNDNSURLSecureSession isTokenExist] + 40
2  Pendo                          0x13adc -[PNDNSURLSecureSession checkAndAcquireToken] + 24
3  Pendo                          0x13600 -[PNDNSURLSecureSession performAuthHandledRequest:completion:] + 60
4  Pendo                          0x114954 -[PNDAnalyticsMessageManager performRequest:block:] + 276
5  Pendo                          0x113e28 -[PNDAnalyticsMessageManager sendAnalyticsEventsBufferToServer:completion:] + 88
6  Pendo                          0x1135f8 -[PNDAnalyticsMessageManager sendAnalytics:] + 140
7  Pendo                          0x113414 -[PNDAnalyticsMessageManager handleBuffer:analytics:] + 736
8  Pendo                          0xbb4cc -[PNDDiskBuffer prepareObjectsForProcessing] + 116
9  libdispatch.dylib              0x213c _dispatch_call_block_and_release + 32
10 libdispatch.dylib              0x3dd4 _dispatch_client_callout + 20
11 libdispatch.dylib              0x15a6c _dispatch_root_queue_drain + 864
12 libdispatch.dylib              0x1609c _dispatch_worker_thread2 + 156
13 libsystem_pthread.dylib        0x48f8 _pthread_wqthread + 228
14 libsystem_pthread.dylib        0x10cc start_wqthread + 8
```

**Comment by TehilaTaub** [comment]
Hi @msrutek-paylocity 
Could you please add full crash log?
How often is this is happening?
We have added that to our backlog and we are looking into this.

**Comment by msrutek-paylocity** [comment]
Hello,

>Could you please add full crash log?

This is currently all we've got.

> How often is this is happening?

This is the first occurrence.

> We have added that to our backlog and we are looking into this.

Thanks!

**Comment by TehilaTaub** [comment]
Hi @msrutek-paylocity 
We added a fix in our next release - 3.3.0, which will be available in early July.
Please reopen if needed.

**Comment by orendayan** [comment]
@msrutek-paylocity 

We've released v3.3.1, please upgrade and let us know if it solved the issues 
please reopen if needed
